src/compas_fab/robots/targets.py

```diff
@@ -490,28 +490,30 @@ class Waypoints(Data):
     Waypoints are useful for tasks like painting, welding, or 3D printing, where the programmer
     wants to define the waypoints the robot should pass through.
 
-    Waypoints are intended to be used for motion planning with a planning backend by using :meth:`compas_fab.robot.plan_motion_with_waypoints`.
-    Different backends might support different types of waypoints.
+    Waypoints are intended to be used for motion planning with a planning backend by using :meth:`compas_fab.robot.plan_cartesian_motion`.
+    Note that different backends support different types of waypoints.
     The method of interpolation between the waypoints is controlled by the motion planner backend.
 
     Attributes
     ----------
     name : str , optional, default = 'target'
         A human-readable name for identifying the target.
+    tool_coordinate_frame : :class:`compas.geometry.Frame` or :class:`compas.geometry.Transformation`, optional
+        The tool tip coordinate frame relative to the flange of the robot.
 
     See Also
     --------
     :class:`PointAxisWaypoints`
     :class:`FrameWaypoints`
     """
 
-    def __init__(self, name="Generic Waypoints"):
+    def __init__(self, tool_coordinate_frame, name="Generic Waypoints"):
         super(Waypoints, self).__init__()
         self.name = name
-
-    @property
-    def __data__(self):
-        raise NotImplementedError
+        # If the user provides a transformation, convert it to a Frame
+        if isinstance(tool_coordinate_frame, Transformation):
+            tool_coordinate_frame = Frame.from_transformation(tool_coordinate_frame)
+        self.tool_coordinate_frame = tool_coordinate_frame
```

Comment:
This is problematic because it's redundant information: the tool has its frame defined, but this will repeat the same data, losing track of its source, so if the tool changes frame, the target has no way to know the data is incorrect.

Reply:
#435